### PR TITLE
Proposition for clickhouse URI driver properties

### PR DESCRIPTION
**Pull request by oev81**
Did not tested. Just a proposition

### CODE REVIEW

File: src/main/java/com/metriql/warehouse/clickhouse/ClickhouseWarehouse.kt [hunk 1]
```diff
@@ -29,7 +29,8 @@ object ClickhouseWarehouse : Warehouse<ClickhouseConfig> {
         val password: String,
         val ssl: Boolean?,
         val usePool: Boolean,
-        val connectionParameters: Map<String, String>?
+        val connectionParameters: Map<String, String>?,
+        val driverUriParameters: Map<String, String>?,
```
Comment (oev81) [inline src/main/java/com/metriql/warehouse/clickhouse/ClickhouseWarehouse.kt:33]: @buremba driverUriParameters can be passed as "driver_uri_parameters" or "driverUriParameters" need tp be used?

File: src/main/java/com/metriql/warehouse/clickhouse/ClickhouseDataSource.kt [hunk 2]
```diff
@@ -26,7 +26,15 @@ class ClickhouseDataSource(override val config: ClickhouseWarehouse.ClickhouseCo
...
-        properties["jdbcUrl"] = "jdbc:clickhouse://${config.host}:$port/${config.database}"
+
+        var queryParams = ""
+        config.driverUriParameters?.let {
+            val list = ArrayList<String>()
+            it.map { (k, v) -> list.add("${k}=${v}") }
+            queryParams = list.joinToString(prefix = "?", separator = "&")
+        }
+        properties["jdbcUrl"] = "jdbc:clickhouse://${config.host}:$port/${config.database}${queryParams}"
+
```
Comment (oev81) [inline src/main/java/com/metriql/warehouse/clickhouse/ClickhouseDataSource.kt:37]: Seems like, In order to pass socket_timeout, a query parameter must be used.

File: src/main/java/com/metriql/warehouse/clickhouse/ClickhouseWarehouse.kt (hunk 1 above)
Comment (buremba) [reply src/main/java/com/metriql/warehouse/clickhouse/ClickhouseWarehouse.kt:33]: They can be passed as `driver_uri_parameters `

File: src/main/java/com/metriql/warehouse/clickhouse/ClickhouseDataSource.kt (hunk 2 above)
Comment (buremba) [reply src/main/java/com/metriql/warehouse/clickhouse/ClickhouseDataSource.kt:37]: Yes but the codebase automatically generates the query parameters and adds them automatically. Looks like the issue is different, I will check again and update the PR with my comment.

**Comment by buremba** [comment]
It looks like the issue was about not having `dataSource.` prefix in the populated properties according to [Hikari docs](https://github.com/brettwooldridge/HikariCP#rocket-initialization). It's not fixed in master.

**Pull request closed without merging**